### ISSUE TITLE
Can't edit Tiddlers in Firefox 57

### ISSUE THREAD

**Issue by Ddhuet** [tiddlydrive/tiddlydrive.github.io#4]
When I try to edit a Tiddler I get:

```
Internal JavaScript Error
Well, this is embarrassing. It is recommended that you restart TiddlyWiki by refreshing your browser
NS_ERROR_MALFORMED_URI: 
```

**Comment by LordRatte** [comment]
Hi, @Ddhuet. Do you mind replying with a list of plugins you have installed?

**Comment by Ddhuet** [comment]
In TiddlyWiki? None.

In FireFox: OpenH264 Video Codec, Widevine Content Decryption Module, and Shockwave Flash.

**Comment by LordRatte** [comment]
Hi, @Ddhuet. I can't seem to recreate the error with those conditions. Please email me an example of a wiki file that is giving you this problem. You can email me at joshua@lordratte.info

**Comment by cosama** [comment]
I have the same error here. Ubunut 16.04, Firefox 57, no extensions/addons really. The error indeed only appears when you try to edit a tiddler. I got the error even with a new TiddlyWiki, once I press new tiddler the error starts popping up and it starts to missformat the entire wiki.

I installed the plug in through google drive. It works on Fedora in Chrome.

**Comment by gse-cc-git** [comment]
Hello @LordRatte,
First thanks for your work and for sharing !
I have the same issue here: Vanilla TW + Firefox 57 (under Ubuntu). It works with Chrome.

**Comment by cosama** [comment]
Yeah same here, I have been using it in Chrome all week and it worked beautifully.

**Comment by LordRatte** [comment]
Hi, all.

Apologies for only getting back to you now. I have been really busy this year. I have managed to recreate the issue and am working on it.

Also, please note that the website may have trouble with https because gitlab is fixing their custom domain functionality.

**Comment by georgeogoodman** [comment]
FYI, LordRatte, I and other(s) have noted that TiddlyDrive isn't working for us due to a cert issue that I'm guessing is related to the problem with https you noted.

**Comment by LordRatte** [comment]
@georgeogoodman Sorry about that! Gitlab has dated the fix for the 20th of February so I have to wait until then, unfortunately.

**Comment by georgeogoodman** [comment]
Thanks for the update, @LordRatte.I'll have to practice patience until then!

**Comment by neongreen** [comment]
It's March already, has Gitlab fixed the bug or do we have to wait for something else?

**Comment by LordRatte** [comment]
Hi, all. Just to clarify: there were two separate problems discussed in this issue.

The original one was about TiddlyDrive not working in Firefox 57. With regards to that, it seems to be an issue with how TiddlyWiki interacts with Firefox in an iframe. It happens regardless of if it's in TiddlyDrive or not. I have submitted a [bug report](https://github.com/Jermolene/TiddlyWiki5/issues/3192) with the owners of the TiddlyWiki repo bringing it to their attention.
In the mean time I can look at a workaround that will only kick in on Firefox. This will not have all the features, such as that of having a correct permanent url. Before that workaround, you'll have to use Chrome.

The problem discussed later was about Gitlab having a certificate issue on their hosted static sites. This affected people in most browsers. This was fixed on the 24th of February. See [this](https://github.com/tiddlydrive/tiddlydrive.github.io/issues/6) issue.

**Comment by neongreen** [comment]
Oh, I see. I thought the `NS_ERROR_MALFORMED_URI:` was because of the certificate.

**Comment by LordRatte** [comment]
Good day, everyone.

I have finally managed to find time and a workaround for the Firefox issue. Because I had to use a bit of a hacky method, there are two features that currently won't be there that are there in Chrome.

1. Updating the address bar with a permalink/permaview
2. Dynamically changing the page title when the Tiddler's title changes

Other than that, it should all be functional again. I would appreciate confirmation from those in this issue that were experiencing this issue (particularly @Ddhuet).

Thanks so much for your patience, all.

Regards. Joshua.

**Comment by cosama** [comment]
Hi Joshua,

I just checked it with the newest version of Firefox (59.0.2) on Ubuntu 16.04 and I could not reproduce the error anymore.

I will still be sticking with chrome for a while (for other reasons than this) but I'm very happy to hear that this works again for when I will be moving back.

Thanks for all your work. Great plugin.
